### PR TITLE
re-organize bottlerocket-specific things

### PR DESCRIPTION
**Pull request by webern**
**Issue number:**

N/A but possibly a precursor to #46

**Description of changes:**

It may be hard to follow the diff but basically we are creating three directories/crates. (Note, however, that we will not move `testsys` into the Bottlerocket directory currently because there are too many PRs going on involving it. Once there are no PRs open on the testsys CLI, I will move that into the directory as shown below).

```
bottlerocket/agents
bottlerocket/testsys
bottlerocket/types
```

`bottlerocket/types` is for common code, like the configs used by both the CLI and the agents.
Moving the CLI into a `bottlerocket` directory should make it more clear that this is for tight integration with Bottlerocket. Basically the `bottlerocket` directory are the parts that could end up being in the Bottlerocket mono-repo (or else interdependent with it)

**Testing done:**

None just yet, but the compiler is happy.

**Terms of contribution:**

By submitting this pull request, I agree that this contribution is dual-licensed under the terms of both the Apache License, version 2.0, and the MIT license.

### CODE REVIEW

**Comment by webern** [comment]
^ fixes

**Review by ecpullen — APPROVED** [review]

**Review by etungsten — APPROVED** [review]
